LiteCore/Query/SQLiteKeyStore+Indexes.cc

```diff
@@ -89,9 +89,10 @@ namespace litecore {
     bool SQLiteKeyStore::createIndex(const IndexSpec& spec, const string& sourceTableName,
                                      Array::iterator& expressions) {
         Assert(spec.type != IndexSpec::kFullText && spec.type != IndexSpec::kVector);
-        QueryTranslator qp(db(), "", sourceTableName);
-        qp.writeCreateIndex(spec.name, sourceTableName, (FLArrayIterator&)expressions, spec.where(),
-                            (spec.type != IndexSpec::kValue));
+        string          hexName{spec.type == IndexSpec::kArray ? litecore::hexName(sourceTableName) : ""};
```

Comment:
Why not use `sourceTableName` as the alternate value instead of `""`? Then you don't have to have conditionals in the following two lines.

Reply:
changed to use optional.